src/lib/scanner.ts

```diff
@@ -98,116 +104,63 @@ export class ScanRunner {
      * Creates the arguments for an execution of {@code sf scanner run}.
      * @param targets The files to be scanned.
      */
-    private async createPathlessArgArray(targets: string[]): Promise<string[]> {
-        const args: string[] = [
-            'scanner', 'run',
-            '--target', `${targets.join(',')}`,
-            '--engine', 'pmd,retire-js',
-            '--json'
-        ];
+    private async createPathlessArgArray(targets: string[]): Promise<Inputs> {
+        const args: Inputs = {
+            'targets': `${targets.join(',')}`,
+            'engine': 'pmd,retire-js',
+            'format': 'json'
+        }
         const customPmdConfig: string = SettingsManager.getPmdCustomConfigFile();
         // If there's a non-null, non-empty PMD config file specified, use it.
         if (customPmdConfig && customPmdConfig.length > 0) {
             if (!(await exists(customPmdConfig))) {
                 throw new Error(messages.error.pmdConfigNotFoundGenerator(customPmdConfig));
             }
-            args.push('--pmdconfig', customPmdConfig);
+            args['pmdconfig'] = customPmdConfig;
         }
         return args;
     }
 
     /**
-     * Uses the provided arguments to run a Salesforce Code Analyzer command.
      * @param args The arguments to be supplied
      */
-    private async invokeAnalyzer(args: string[]): Promise<ExecutionResult> {
-        return new Promise((res) => {
-            const cp = cspawn.spawn('sf', args);
-
-            let stdout = '';
-
-            cp.stdout.on('data', data => {
-                stdout += data;
-            });
+    private async invokeAnalyzer(args: Inputs): Promise<AnyJson> {
+        const sfVersion = '0.0.5';
```

Comment:
@jfeingold35 @stephen-carter-at-sf  - Where is this version used? I could instead read this from package.json or have it set in the settings. But I'm still trying to understand the significance of this.

Reply:
Yeah this I really dislike. The "sf" cli version is something that is added to the RunOptions (see line 86 of InputProcessor) that we ultimately send off in our telemetry calls (see line 147 and 182 of DefaultRuleManager) as part of our telemetry data.

That is, our rule manager assumes that we are running from the CLI which is why this artifact still exists. You can probably just put in x.x.x if you want - again it's only for telemetry data.

Reply:
Keeping it as 0.5.0 for now.

Reply:
Done.